src/map_field.cpp

```diff
@@ -1499,8 +1497,13 @@ void map::player_in_field( Character &you )
                     you.add_msg_player_or_npc( m_bad, _( "The acid burns your body!" ),
                                                _( "The acid burns <npcname>'s body!" ) );
                 } else if( total_damage > 0 ) {
-                    you.add_msg_player_or_npc( m_bad, _( "The acid burns your legs and feet!" ),
-                                               _( "The acid burns <npcname>'s legs and feet!" ) );
+                    if( you.has_effect( effect_quadruped_full ) || you.has_effect( effect_quadruped_half ) ) {
+                        you.add_msg_player_or_npc( m_bad, _( "The acid burns your hands and feet!" ),
+                                                   _( "The acid burns <npcname>'s hands and feet!" ) );
+                    } else {
+                        you.add_msg_player_or_npc( m_bad, _( "The acid burns your legs and feet!" ),
+                                                   _( "The acid burns <npcname>'s legs and feet!" ) );
+                    }
```

Comment:
Can you change these messages to drop the effect check? We don't want random checks like this scattered throughout the code, we want it to basically be in one place (right now that's get_ground_contact_bodyparts).

Reply:
So, perhaps add a `ground_contact_body_part_types` like the `get_ground_contact_body_parts`, that does the quadruped check and the quadruped check just calls through to that and generates a vector of bodypart_id corresponding.

You can then use that and the name of the body part type to generate these style messages.

https://github.com/CleverRaven/Cataclysm-DDA/blob/64852a6f60a6959776f9426fcd1e927b1af03a0e/src/bodypart.h#L248-L256

Reply:
With `enumerate_as_string` to concatenate the names.

Reply:
No, because it'll be the name of the type, not the name of all the parts in the type. I would recommend you do it this way because it'll be easier to translate (and translate correctly) than your current solution is.